image.go

```diff
@@ -141,22 +142,27 @@ func Fit(buf []byte, o ImageOptions) (Image, error) {
 		fitHeight = &o.Width
 	}
 
-	// if input ratio > output ratio
-	// (calculation multiplied through by denominators to avoid float division)
-	if originWidth*(*fitHeight) > (*fitWidth)*originHeight {
-		// constrained by width
-		*fitHeight = *fitWidth * originHeight / originWidth
-	} else {
-		// constrained by height
-		*fitWidth = *fitHeight * originWidth / originHeight
-	}
+	*fitHeight, *fitWidth = calculateDestinationFitDimension(originWidth, originHeight, *fitWidth, *fitHeight)
 
 	opts := BimgOptions(o)
 	opts.Embed = true
 
 	return Process(buf, opts)
 }
 
+// calculateDestinationFitDimension calculates the fit area based on the image and desired fit dimensions
+func calculateDestinationFitDimension(imageWidth, imageHeight, fitWidth, fitHeight int) (int, int) {
+	if imageWidth*fitHeight > fitWidth*imageHeight {
+		// constrained by width
+		fitHeight = int(math.Round(float64(fitWidth) * float64(imageHeight) / float64(imageWidth)))
```

Comment:
in bimg I saw often using int(math.Floor(f + 0.5)) instead of int(math.Round(f)), not sure why

Reply:
found an explanation in https://golang.org/src/math/floor.go: math.Floor have (architecture-specific) implementations in assembler (e.g. in https://golang.org/src/math/floor_386.s), math.Round not. seems they prefer math.Floor because of the higher performance.

Reply:
I would still use math.Round for better readability

Reply:
Thanks for the sanity check! I think math.Round() is better as well. Performance in this section is hardly a bottleneck.

Reply:
I wanted also check if we should use math.Ceil rather than math.Round, I took a peek in libvips but I didnt come yet to a result. or we could play with vipsthumbnail if e.g. a case with 411.4 px will be rounded up (-> use math.Ceil) or rounded down (-> use math.Round)

Reply:
libvips uses Rounding, when scaling a 900x500 (a nice fractional conversion size) image to 312 outer dimensions, it rounds to 173 instead of 174.

- 312 / (900/500) = 173.3
- 313 / (900/500) = 173.8

The command
```
vipsthumbnail ./james-ree-1312004-unsplash.jpg -s 312x312 -o tn_%s.jpg --vips-cache-trace
```
produces a `312x173` size.

![image](https://user-images.githubusercontent.com/39014/51528195-6ef30b80-1e36-11e9-872c-e6f5d7497b9a.png)
_Image credits: [James Ree](https://unsplash.com/@jrmotion), original:  https://unsplash.com/photos/nM7djMZfSuo_